### ISSUE TITLE
GUI is "too big" for small screens.

### ISSUE THREAD

**Issue by mayuso** [demilich1/metastone#84]
I am not sure if small screens are still used a lot(Small in this case is 1366x768 or smaller), but I do. 
I just opened MetaStone on my laptop, and I can´t see the bottom part of the GUI (Donate button, start button on simulation mode.....and the rest).

I tried making the window resizeable, but the elements do not resize, so that is not a viable option right now.

Any ideas on how to fix this?

**Comment by demilich1** [comment]
I am afraid there is no simple fix. I opted for a fixed size window because it gave me guarantees for the layout. Resizable windows are much more user-friendly though, I should have taken that more seriously. Still, it can be fixed, it just is quite a lot of work. I am still dreaming about that UI artist who stumbles upon this project and decides to overhaul and redesign the whole UI :P

**Comment by mayuso** [comment]
I have done dozens of UIs with swing and LibGDX, but never used JavaFX (I believe this is JavaFX, not sure). I am going to learn about is as much as I can and (hopefully) improve it in the "near" future.

**Comment by webadict** [comment]
Is there a way to switch from exact sizing to relative sizing? There's a
couple of GUI issues that will probably happen from doing that, but it
would make resizing not too bad. Either that or some sort of INI file that
could change resolution. Also, yeah this is JavaFX.

**Comment by demilich1** [comment]
> Is there a way to switch from exact sizing to relative sizing? 

Well, you can specify relative sizes in .css, but there is no global switch for it. I don't see any solution which does not a require adjustments to basically every UI element.

**Comment by Facon** [comment]
Scroll bars?

**Comment by mayuso** [comment]
I have gained some experience on JavaFX since I created this issue, I am working on this issue on my fork.

**Comment by mayuso** [comment]
Minimum size will be 800x600 for now, seems to work fine on all the windows I have tested. I will wait until I have all windows edited to properly adjust size before I make a PR.

**Comment by mayuso** [comment]
So I tried fixing this some months ago, you guys didn't really like the result.

I have been working on JavaFX a lot this whole time, and I feel like I can do a better job, and I need a challenge. It will probably take a couple weeks or even a month, but no one likes easy challenges :)

I reopen this issue while I work on reworking part of the UI for this to get fixed.

**Comment by webadict** [comment]
Sweet! That'd be nice to have a few of those GUI bugs fixed!

**Comment by mayuso** [comment]
A bit late, I guess, but I started working on this again. I am working on my fork, and I hope to have something worth merging asap.

I'll throw my impressions:

The main problem right now is that metastone uses a fixed size/resolution, this isn't a big problem in big screens (1600x900 or higher), but it is a problem in smaller ones (Like my laptops 1366x768), some windows load buttons outside of the screen, so I can't reach them.

I am currently working on making each element resizeable so everything scales with window size, it's looking nice, but giving the user the option to resize doesn't look like a good idea: Elements resize properly when changing the window size, but it also kinda destroys alignment in some windows, so yeah...not ideal.

I feel like having a bunch of default/premade resolutions AND not allowing the user to manually resize the window might be the way to go here. Having 5, 10 or 15 default resolutions is easily maintainable (Specially with JavaFX).

*Offtopic:* I just found out about issue #7, I already throwed myself into FXML edition, so i'll also deal with this long time ignored issue too, shouldn't be a problem.

I am leaving `primaryStage.setResizable(false);` so I am sure it won't be a problem, the default user experience will remain exactly the same (At least for now), but I'll need people to test the changes, because I don't have larger screens.
